protobuf-lite/src/main/java/io/grpc/protobuf/lite/ProtoLiteUtils.java

```diff
@@ -107,23 +110,38 @@ public T parse(InputStream stream) {
             }
           }
         }
+        CodedInputStream cis;
+        if (stream instanceof ReadableBuffer) {
+          // This is only a temporary solution, and deserves a better solution.  We copy here to
+          // avoid Protobuf's bad deserialization buffering.  In reality, we should be able to pass
+          // a collection of ByteBuffers to the decoder and avoid buffering at all.  The code below
+          // should eventually be removed in favor of a proper solution.
+          ReadableBuffer buffer = (ReadableBuffer) stream;
+          // copyBuffer must never leave scope and may never be modified.
+          ByteBuffer copyBuffer = ByteBuffer.allocate(buffer.readableBytes());
+          buffer.readBytes(copyBuffer);
+          // Unsafe is used here because we are very careful about how we use it.  It must not be
+          // used without careful scrutiny.
+          cis = UnsafeByteOperations.unsafeWrap(copyBuffer).newCodedInput();
```

Comment:
> It gets access (in a roundabout way) to setting the protected bufferIsImmutable factory flag.

But are we actually using that? Based on another comment it seems like we aren't:

> \4. Copy from "the buffer" reading out into a ByteString or byte array to read out the field from the proto.

> The after case also opens up the possibility of using aliasing, which would avoid the copy in step 4 in both cases (but may have other side effects).